test/unit/util/event_handler_error.dart

```diff
@@ -21,27 +21,27 @@ import 'package:react/react_client.dart' show ReactElement;
 import 'package:react_testing_library/react_testing_library.dart' as rtl;
 import 'package:test/test.dart';
 
+import 'exception.dart';
 import 'over_react_stubs.dart';
 
 void testEventHandlerErrors(
     List<String> events, void Function(Element element) userEventTrigger, dynamic elementTypeToRender,
     {dynamic children, Map<String, dynamic> additionalProps = const {}}) {
+      const targetTestId = 'test-subject-11';
+      throwingHandler(event) => throw ExceptionForTesting('Mission Failed');
+      buildEventHandlers() => {for (final value in events) value: throwingHandler};
+      buildProps() => {
+            ...buildEventHandlers(),
+            ...additionalProps,
+            ...{defaultTestIdKey: targetTestId}
+          };
+      final targetElement = react
+          .div({}, children == null ? elementTypeToRender(buildProps()) : elementTypeToRender(buildProps(), children));
```

Comment:
This isn't quite what I meant 😄.

Instead of being executed in the nested group, now it's being executed in its parent group. Could we move it to a function or a `setUp` so it's only run when the tests are run, as opposed to when the tests are being declared?

Reply:
Im not sure i understand. Why does it matter if it is when the test is declared vs when its run? It's a final variable, it's the same thing in both tests, it doesn't need to be run before every test as setUp would do, in fact it would then not be final anymore and would be rebuilding the same thing before every test. Why is that better?

Reply:
Sorry, I wasn't doing a good job of explaining why I made this suggestion.

In general, we should avoid doing computations outside of test blocks because:
1. The code in `group` blocks gets run at startup every time you run tests, even if you're running a subset of the tests (by test name filtering via `-n`/`-N`, tag filtering, etc.). The `group`s themselves aren't skipped and all of them get run, it's just the `test` blocks and corresponding setUp/etc. blocks that get skipped. So, anything that's done in `group`s adds overhead to test startup, and also can get run unnecessarily when tests are being filtered.

    This one instance of creating a ReactElement isn't a big deal, but if we get in the habit of doing it in more places and with more expensive computations, it can add up and slow things down.

1. Related, if something goes wrong when initializing that variable, it's nicer if it gets run as part of the test it's being used in, so that the error message can be more easily associated with the test.


So I think either making targetElement a function instead of a variable would probably be best here.

Reply:
Oohhh KK that makes so much sense. Thank you for clarifying! I never knew that this was the case and for sure do not want that to become an issue. I will remember this lesson going forward :)